source/agora/consensus/PreImage.d

```diff
@@ -392,13 +394,30 @@ public struct PreImageCycle
             this.index = seek_index;
             const cycle_seed = hashMulti(
                 this.secret, "consensus.preimages", 0);
-            this.seeds.reset(cycle_seed);
-            this.preimages.reset(this.seeds.byStride[$ - 1 - this.index]);
+
+            auto seed = getWellKnownPreImageSeed(this.secret, PreImageCycle.PreImageCount,
+               this.cycles, this.index, this.seeds.interval);
+            if (seed == Hash.init)
+            {
+                this.seeds.reset(cycle_seed);
+                this.preimages.reset(this.seeds.byStride[$ - 1 - this.index]);
+            }
+            else
+            {
+                this.preimages.reset(seed);
+            }
+
```

Comment:
This is why it is quite problematic, in the end, to have WK preimages.
The code here is test-specific but gives no indication of it. Dependency injection should be done via interfaces (in the broad sense, not `interface`), not be adding special cases.

Reply:
Would it be better to add `version (unittest)`?